spring-cloud-app-starter-metadata-maven-plugin/src/main/java/org/springframework/cloud/stream/app/documentation/plugin/MetadataAggregationMojo.java

```diff
@@ -139,6 +140,15 @@ private Result(ConfigurationMetadata metadata, Properties whitelist) {
 							whiteList = merge(whiteList, is);
 						}
 					}
+					else {
+						File backupLocalWhiteList = new File(file, BACKUP_WHITELIST_PATH);
+						if (backupLocalWhiteList.canRead()) {
+							try (InputStream is = new FileInputStream(backupLocalWhiteList)) {
+								getLog().debug("!!!! Merging whitelist from " + path);
+								whiteList = merge(whiteList, is);
+							}
```

Comment:
Looks like we could factor both `if..else` branches to an utility method with `whiteList` and `file` as parameters.

In other words we have here a duplicate code.

Although that's just my opinion and we can merge this already.

Reply:
@artembilan Thanks, we might remove that check for the backup file once we remove the original whitelist properties file. In that case, the `else` block will be gone.